website/source/docs/guides/containers-guide.md

```diff
@@ -0,0 +1,332 @@
+---
+name: 'Consul with Containers'
+content_length: 15
+id: containers-guide
+layout: content_layout
+products_used:
+  - Consul
+description: HashiCorp provides an official Docker image for running Consul and this guide demonstrates its basic usage.
+level: Implementation
+---
+
+# Consul with Containers
+
+In this guide, you will learn how to deploy two, joined Consul agents each running in separate Docker containers. You will also register a service and perform basic maintenance operations. The two Consul agents will form a small datacenter. 
+
+By following this guide you will learn how to:
+
+1. Get the Docker image for Consul
+1. Configure and run a Consul server
+1. Configure and run a Consul client
+1. Interact with the Consul agents
+1. Perform maintenance operations (backup your Consul data, stop a Consul agent, etc.)
+
+The guide is Docker-focused, but the principles you will learn apply to other container runtimes as well. 
+
+!> Security Warning This guide is not for production use. Please refer to the [Consul Reference Architecture](https://learn.hashicorp.com/consul/datacenter-deploy/reference-architecture) for Consul best practices and the [Docker Documentation](https://docs.docker.com/) for Docker best practices.
+
+## Prerequisites
+
+### Docker
+
+You will need a local install of Docker running on your machine for this guide. You can find the instructions for installing Docker on your specific operating system [here](https://docs.docker.com/install/).
+
+### Consul (Optional)
+
+If you would like to interact with your containerized Consul agents using a local install of Consul, follow the instructions [here](https://www.consul.io/docs/install/index.html) and install the binary somewhere on your PATH.
+
+## Get the Docker Image
+
+First, pull the latest image. You will use Consul's official Docker image in this guide.
+
+```sh
+$ docker pull consul
+```
+
+Check the image was downloaded by listing Docker images that match `consul`.
+
+```sh
+$ docker images -f 'reference=consul'
+REPOSITORY     TAG      IMAGE ID        CREATED             SIZE
+consul         latest   c836e84db154     4 days ago         107MB
+```
+## Configure and Run a Consul Server
+
+Next, you will use Docker command-line flags to start the agent as a server, configure networking, and bootstrap the cluster when one server is up.
+
+```sh
+$ docker run \
+    -d \
+    -p 8500:8500 \
+    -p 8600:8600/udp \
+    --name=badger \
+    consul agent -server -ui -node=server-1 -bootstrap-expect=1 -client=0.0.0.0
+```
+
+Since you started the container in detached mode, `-d`, the process will run in the background. You also set port mapping to your local machine as well as binding the cient interface of our agent to 0.0.0.0. This allows you to work directly with the Consul cluster from your local machine and to access Consul's UI and DNS over localhost. Finally, you are using Docker's default bridge network.
```

Comment:
```suggestion
Since you started the container in detached mode, `-d`, the process will run in the background. You also set port mapping to your local machine as well as binding the client interface of our agent to 0.0.0.0. This allows you to work directly with the Consul datacenter from your local machine and to access Consul's UI and DNS over localhost. Finally, you are using Docker's default bridge network.
```